src/test/java/com/trustly/api/NotificationsTest.java

```diff
@@ -0,0 +1,178 @@
+
+      Assertions.assertEquals(200, status.get());
+      Assertions.assertNotNull(responseString.get());
+    }
+  }
+
+  @Test
+  void testUnknownNotification() throws Exception {
+
+    try (TrustlyApiClient client = new TrustlyApiClient(settings, new NoOpJsonRpcSigner())) {
+
+      final AtomicReference<Object> receivedUnknownValue = new AtomicReference<>();
+
+      client.addOnUnknownNotification(args -> {
+        args.respondWithOk();
+        receivedUnknownValue.set(args.getData().getAny().get("something"));
+      });
+
+      final InputStream is = this.getClass().getResourceAsStream("/notifications/incoming/_unknown.json");
+      final Map<String, String> headers = new HashMap<>();
+      final AtomicInteger status = new AtomicInteger();
+      final AtomicReference<String> responseString = new AtomicReference<>();
+
+      final NotificationResponder responder = new NotificationResponder() {
+        @Override
+        public void addHeader(String key, String value) {
+          headers.put(key, value);
+        }
+
+        @Override
+        public void setStatus(int httpStatus) {
+          status.set(httpStatus);
+        }
+
+        @Override
+        public void writeBody(String value) {
+          responseString.set(value);
+        }
+      };
+
+      TrustlyApiClientExtensions.handleNotificationRequest(is, responder);
+
+      Assertions.assertEquals(200, status.get());
+      Assertions.assertEquals("application/json", headers.get("Content-Type"));
+      Assertions.assertNotNull(responseString.get());
+      Assertions.assertEquals("abc", receivedUnknownValue.get());
+    }
+  }
+
+  @Test
+  void testCancelNotification() throws Exception {
+
+    try (TrustlyApiClient client = new TrustlyApiClient(settings)) {
+
+      final AtomicInteger receivedNotificationDataCounter = new AtomicInteger();
+
+      client.addOnCancelListener(args -> {
+
+        receivedNotificationDataCounter.incrementAndGet();
+        args.respondWithOk();
+      });
+
+      final InputStream is = this.getClass().getResourceAsStream("/notifications/incoming/cancel.json");
+      final Map<String, String> headers = new HashMap<>();
+      final AtomicInteger status = new AtomicInteger();
+      final AtomicReference<String> responseString = new AtomicReference<>();
+
+      final NotificationResponder responder = new NotificationResponder() {
+        @Override
+        public void addHeader(String key, String value) {
+          headers.put(key, value);
+        }
+
+        @Override
+        public void setStatus(int httpStatus) {
+          status.set(httpStatus);
+        }
+
+        @Override
+        public void writeBody(String value) {
+          responseString.set(value);
+        }
+      };
+
+      TrustlyApiClientExtensions.handleNotificationRequest(is, responder);
+
+      Assertions.assertEquals(200, status.get());
+      Assertions.assertNotNull(responseString.get());
+    }
+  }
```

Comment:
Why have a specific test for `Cancel` when it is included in the first test?

Reply:
That one is with signature verification, the others are not